src/components/AddXpubDialog/AddXpubDialog.tsx

```diff
@@ -0,0 +1,102 @@
+import { CirclePlus } from 'lucide-react';
+import { Button } from '@/components/ui/button.tsx';
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog.tsx';
+import { Label } from '@/components/ui/label.tsx';
+import { Input } from '@/components/ui/input.tsx';
+import React, { useEffect, useState } from 'react';
+import { HD } from '@bsv/sdk';
+import { errorWrapper } from '@/utils';
+import { toast } from 'sonner';
+import { useDebounce } from 'use-debounce';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useSpvWalletClient } from '@/contexts';
+
+export const AddXpubDialog = () => {
+  const [xPriv, setXPriv] = useState<string>('');
+  const [xPub, setXPub] = useState<string>('');
+  const [debouncedXPriv] = useDebounce(xPriv, 500);
+  const queryClient = useQueryClient();
+
+  const { spvWalletClient } = useSpvWalletClient();
```

Comment:
Could we implement the `useSpvWalletClient` (or similar custom hook) which returns the client - without `undefined` |`null` posibility?

As far as understand it, for standard usage, the client should be always defined, and - thanks to that, we would avoid the `!.` like in `spvWalletClient!.AdminNewXpub`.

Reply:
I was thinking about that. Client creates on login stage. Before that app does not have a key to create a client. So it initialized as null on the start.  I can add if statement to check if spvWalletClient is not null. This should be enough to remove `!.`

Reply:
But I was thinking about specialized `useSpvWalletClient` which is designed to cover the cases when the `client` should be defined - so almost always (except login screen).

Consider it - but if you don't want to do it right now, just resolve this comment 😄

Reply:
The initial state of the client is `null`. To create a client app needs an access key or xPriv. Maybe you have ideas how to create a client without access key or xPriv ?